geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaDataStoreSchemaManager.scala

```diff
@@ -137,6 +138,38 @@ trait KafkaDataStoreSchemaManager extends DataStore with LazyLogging {
     }
   }
 
+  def updateKafkaSchema(typeName: String, sft: SimpleFeatureType) = {
+
+    // Get previous schema and user data
+    val previousSft = getSchema(typeName)
+    val schemaTypeName = sft.getTypeName()
+
+    // Prevent modifying wrong type if type names don't match
+    if (!schemaTypeName.equals(typeName.toString)) {
+      throw new UnsupportedOperationException("Updating the type name of a schema is not allowed " + schemaTypeName + " " + typeName)
+    }
+
+    // Check that unmodifiable user data has not changed
+    val unmodifiableUserdataKeys = Set(SCHEMA_VERSION_KEY, TABLE_SHARING_KEY, SHARING_PREFIX_KEY,
+      DEFAULT_DATE_KEY, ST_INDEX_SCHEMA_KEY, SimpleFeatureTypes.ENABLED_INDEXES)
+
+    unmodifiableUserdataKeys.foreach { case (key) =>
+      if (sft.getUserData.keySet().contains(key) && sft.userData[String](key) != previousSft.userData[String](key)) {
+        throw new UnsupportedOperationException("Updating " + key + " is not allowed")
```

Comment:
[scala] in general prefer using string interpolation: `s"Updating $key is not allowed"`